packages/styles/scss/components/checkbox/_checkbox.scss

```diff
@@ -378,19 +380,29 @@
   .#{$prefix}--checkbox-wrapper--slug
     .#{$prefix}--checkbox-label-text
     .#{$prefix}--ai-label,
-  .#{$prefix}--checkbox-group--slug legend.#{$prefix}--label .#{$prefix}--slug,
-  .#{$prefix}--checkbox-wrapper--slug
+  .#{$prefix}--checkbox-group--decorator
+    legend.#{$prefix}--label
+    .#{$prefix}--ai-label,
+  .#{$prefix}--checkbox-wrapper--decorator
     .#{$prefix}--checkbox-label-text
-    .#{$prefix}--slug {
+    .#{$prefix}--ai-label,
+  .#{$prefix}--checkbox-group--decorator
+    legend.#{$prefix}--label
+    .#{$prefix}--checkbox-group-inner--decorator
+    > *,
+  .#{$prefix}--checkbox-wrapper--decorator
+    .#{$prefix}--checkbox-label-text
+    .#{$prefix}--checkbox-wrapper-inner--decorator
+    > * {
     margin-inline-start: $spacing-03;
   }
 
   .#{$prefix}--checkbox-wrapper--slug
     .#{$prefix}--checkbox-label-text
     .#{$prefix}--ai-label__button--inline,
-  .#{$prefix}--checkbox-wrapper--slug
+  .#{$prefix}--checkbox-wrapper--decorator
     .#{$prefix}--checkbox-label-text
-    .#{$prefix}--slug__button--inline {
+    .#{$prefix}--ai-label__button--inline {
```

Comment:
so since the slug prop is only being deprecated and they could still be using the old slug styles, I believe we need to keep all the slug styles for now, but just add the scenario for decorator with ai-label.

The three scenario's im thinking are:
- deprecated slug prop with slug classes
- deprecated slug prop with ai-label classes
- decorator prop with whatever is inside 

```
.#{$prefix}--checkbox-wrapper--decorator
    .#{$prefix}--checkbox-label-text
    .#{$prefix}--ai-label__button--inline,
.#{$prefix}--checkbox-wrapper--slug
    .#{$prefix}--checkbox-label-text
    .#{$prefix}--ai-label__button--inline,
  .#{$prefix}--checkbox-wrapper--slug
    .#{$prefix}--checkbox-label-text
    .#{$prefix}--slug__button--inline {
```